packages/plugin-core/src/commands/MoveHeader.ts

```diff
@@ -597,12 +602,72 @@ export class MoveHeaderCommand extends BasicCommand<
       origin,
       dest
     );
-    return { ...opts, updated };
+
+    return { ...opts, changed: updated };
   }
 
-  addAnalyticsPayload(_opts?: CommandOpts, out?: CommandOutput) {
-    return {
-      updatedCount: out?.updated.length || 0,
+  trackProxyMetrics({
+    out,
+    noteChangeEntryCounts,
+  }: {
+    out: CommandOutput;
+    noteChangeEntryCounts: {
+      createdCount?: number;
+      deletedCount?: number;
+      updatedCount?: number;
     };
+  }) {
+    const extension = ExtensionProvider.getExtension();
+    const engine = extension.getEngine();
+    const { vaults } = engine;
+
+    // only look at origin note
+    const { origin } = out;
+
+    const headers = _.toArray(origin.anchors).filter((anchor) => {
+      return anchor !== undefined && anchor.type === "header";
+    }) as DNoteHeaderAnchor[];
+
+    const numOriginHeaders = headers.length;
+    const originHeaderDepths = headers.map((header) => header.depth);
+    const maxOriginHeaderDepth = _.max(originHeaderDepths);
+    const meanOriginHeaderDepth = _.mean(originHeaderDepths);
+    const movedHeaders = out.nodesToMove.filter((node) => {
+      return node.type === "heading";
+    }) as Heading[];
+    const numMovedHeaders = movedHeaders.length;
+    const movedHeaderDepths = movedHeaders.map((header) => header.depth);
+    const maxMovedHeaderDepth = _.max(movedHeaderDepths);
+    const meanMovedHeaderDepth = _.mean(movedHeaderDepths);
+
+    AnalyticsUtils.track(EngagementEvents.RefactoringCommandUsed, {
+      command: this.key,
+      ...noteChangeEntryCounts,
+      traits: origin.traits || [],
+      numVaults: vaults.length,
+      numChildren: origin.children.length,
+      numlinks: origin.links.length,
+      numChars: origin.body.length,
+      noteDepth: DNodeUtils.getDepth(origin),
+      numOriginHeaders,
```

Comment:
if we're tracking refactor operations, it will be good to have a type that enumerates what these refactor operations could be and use them when doing analytics across the various refactor commands

Reply:
not enumerating keys but the properties. eg. `numLinks`

Reply:
ah okay, I misunderstood. @jonathanyeung gave the same suggestion, and it's pushed now.